### PR TITLE
feat: [PROD-3845] Add tracer in pubsub

### PR DESCRIPTION
**Pull request by Sansinj**
In this PR:
- Add option to provide TracerProvider in pubsub setup
- Tracer in  Subscriber and Publisher

### CODE REVIEW

File: pubsubx/config.go [hunk 1]
```diff
@@ -22,6 +24,20 @@ type KafkaConfig struct {
 	Brokers []string `json:"brokers"`
 }
 
+type pubSubOptions struct {
+	tracerProvider trace.TracerProvider
```
Comment (etiennecl) [inline pubsubx/config.go:28]: Should be a pointer so that the tracer is optional

File: pubsubx/kafka.go [hunk 2]
```diff
@@ -4,14 +4,16 @@ import (
...
 )
 
 // TODO: add publisher configs
-func SetupKafkaPublisher(l *logrusx.Logger, c *Config) (Publisher, error) {
+func SetupKafkaPublisher(l *logrusx.Logger, c *Config, opts *pubSubOptions) (Publisher, error) {
 	publisher, err := kafka.NewPublisher(
 		kafka.PublisherConfig{
 			Brokers:   c.Providers.Kafka.Brokers,
 			Marshaler: kafka.DefaultMarshaler{},
+			Tracer:    NewOTELSaramaTracer(otelsarama.WithTracerProvider(opts.tracerProvider)),
```
Comment (xWiiLLz) [inline pubsubx/kafka.go:16]: Should check if `opts.tracerProvider` is nil. In that case we shouldn't include the tracer at all

File: pubsubx/kafka.go [hunk 3]
```diff
@@ -23,14 +25,15 @@ func SetupKafkaPublisher(l *logrusx.Logger, c *Config) (Publisher, error) {
...
-func SetupKafkaSubscriber(l *logrusx.Logger, c *Config, group string) (Subscriber, error) {
+func SetupKafkaSubscriber(l *logrusx.Logger, c *Config, opts *pubSubOptions, group string) (Subscriber, error) {
 	saramaSubscriberConfig := kafka.DefaultSaramaSubscriberConfig()
 	saramaSubscriberConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
 
 	conf := kafka.SubscriberConfig{
 		Brokers:               c.Providers.Kafka.Brokers,
 		Unmarshaler:           kafka.DefaultMarshaler{},
 		OverwriteSaramaConfig: saramaSubscriberConfig,
+		Tracer:                NewOTELSaramaTracer(otelsarama.WithTracerProvider(opts.tracerProvider)),
```
Comment (xWiiLLz) [inline pubsubx/kafka.go:36]: Same here

Comment (etiennecl) [inline pubsubx/kafka.go:36]: Check if nil else add it to the config

**Review by xWiiLLz — APPROVED** [review]
Some minor stuff buts otherwise looks good

File: pubsubx/config.go (hunk 1 above)
Comment (Sansinj) [reply pubsubx/config.go:28]: Do I still need to use a pointer considering `trace.TracerProvider` is an interface?

Comment (Sansinj) [reply pubsubx/config.go:28]: Also I could use type otelx.TracerProvider instead, but my first guess is that it's not a good idea.

**Review by etiennecl — APPROVED** [review]

**Pull request merged**